### ISSUE TITLE
`env.spec.make` makes a different benchmark

### ISSUE THREAD

**Issue by uduse** [facebookresearch/CompilerGym#587]
## 🐛 Bug

```python
import compiler_gym
env = compiler_gym.make('llvm-v0')
env.reset(benchmark="cbench-v1/bitcount")
assert env.benchmark  == env.spec.make().benchmark
```

This raises `AssertionError`.

Honestly I feel like this is more of a design problem: `env.spec` doesn't actually specify specification because things like `env.reset` can mutate specification. They could mutate specification because that's the only way for users to recycle backend services. However, users shouldn't be worry about managing services life-times. In other words, if we free users from managing backend services (using [something like this](https://github.com/facebookresearch/CompilerGym/issues/572#issuecomment-1055758719)), we could remove APIs like `env.reset` and `env.benchmark = ...`. Then `env.spec` could reliably specifies environment instances. Just my two cents.

## Environment

- CompilerGym: 0.2.2
- OS: Linux
- Python version: 3.8

**Comment by ChrisCummins** [comment]
Yep, this is a bug. The `EnvSpec._kwargs` dict needs to be updated by the setters for benchmark, observation space, and reward space. Thanks for the report @uduse!

Interestingly, as of [head](https://github.com/openai/gym/blob/master/gym/envs/registration.py#L104), `EnvSpec.kwargs` is now public, see #588.

Cheers,
Chris
